state/executor.go

```diff
@@ -0,0 +1,229 @@
+package state
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"time"
+
+	abci "github.com/lazyledger/lazyledger-core/abci/types"
+	tmstate "github.com/lazyledger/lazyledger-core/proto/tendermint/state"
+	"github.com/lazyledger/lazyledger-core/proxy"
+	lltypes "github.com/lazyledger/lazyledger-core/types"
+
+	abciconv "github.com/lazyledger/optimint/conv/abci"
+	"github.com/lazyledger/optimint/hash"
+	"github.com/lazyledger/optimint/log"
+	"github.com/lazyledger/optimint/mempool"
+	"github.com/lazyledger/optimint/types"
+)
+
+type BlockExecutor struct {
+	proxyApp proxy.AppConnConsensus
+	mempool  mempool.Mempool
+
+	logger log.Logger
+}
+
+func NewBlockExecutor(mempool mempool.Mempool, proxyApp proxy.AppConnConsensus, logger log.Logger) *BlockExecutor {
+	return &BlockExecutor{
+		proxyApp: proxyApp,
+		mempool:  mempool,
+		logger:   logger,
+	}
+}
+
+func (e *BlockExecutor) CreateProposalBlock(height uint64, commit *types.Commit, state State) *types.Block {
```

Comment:
We might want another name for this. Proposal is a consensus message (the thing that optimint does not do).